internal/cmd/oidc/oidc.go

```diff
@@ -0,0 +1,183 @@
+package oidc
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/kyma-project/cli.v3/internal/clierror"
+	"github.com/kyma-project/cli.v3/internal/cmdcommon"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+)
+
+type oidcConfig struct {
+	*cmdcommon.KymaConfig
+
+	output              string
+	caCertificate       string
+	clusterServer       string
+	audience            string
+	IDTokenRequestURL   string
+	IDTokenRequestToken string
+}
+
+type TokenData struct {
+	Count int    `json:"count"`
+	Value string `json:"value"`
+}
+
+func NewOIDCCMD(kymaConfig *cmdcommon.KymaConfig) *cobra.Command {
+	cfg := oidcConfig{
+		KymaConfig: kymaConfig,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "oidc",
+		Short: "Create kubeconfig with an OIDC token",
+		Long:  "Create kubeconfig with an OIDC token generated with a Github Actions token",
+		PreRun: func(_ *cobra.Command, args []string) {
+			clierror.Check(cfg.complete())
+			clierror.Check(cfg.validate())
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			clierror.Check(runOIDC(&cfg))
+		},
+	}
+
+	cmd.Flags().StringVar(&cfg.output, "output", "", "Path to the output kubeconfig file")
+	cmd.Flags().StringVar(&cfg.caCertificate, "ca-certificate", "", "Path to the CA certificate file")
+	cmd.Flags().StringVar(&cfg.clusterServer, "cluster-server", "", "URL of the cluster server")
+	cmd.Flags().StringVar(&cfg.audience, "audience", "", "Audience of the token")
+	cmd.Flags().StringVar(&cfg.IDTokenRequestURL, "id-token-request-url", "", "URL to request the ID token")
```

Comment:
I would suggest to add information about defaulting (by default value is get from env)